src/db.rs

```diff
@@ -246,16 +246,64 @@ impl DBStore {
         mode: rocksdb::IteratorMode,
         mut filter_fn: F,
     ) -> impl Iterator<Item = [u8; N]> + '_ {
-        self.db
-            .iterator_cf_opt(cf, readopts, mode)
-            .filter_map(move |row| {
-                let k = &*row.expect("cf iterator failed").0;
-                if filter_fn(k) {
-                    Some(k.try_into().unwrap())
+        struct Iter<F>(F);
+
+        impl<F: FnMut() -> Option<Item>, Item> Iterator for Iter<F> {
+            type Item = Item;
+
+            fn next(&mut self) -> Option<Self::Item> {
+                self.0()
+            }
+        }
```

Comment:
Interesting hack though perhaps it should be explicit.

Reply:
when looking at it now this looks exactly like core::iter::from_fn, will change this, the hack will be behind the documentation of the standard lib, explicit enough? :)

Reply:
I meant making a struct with explicit fields. But using `from_fn` from `std` is tempting, since that reduces boilerplate.